tests/optimisation_test/recipe_converter.py

```diff
@@ -3,58 +3,84 @@
 from archemist.core.persistence.yaml_handler import YamlHandler
 from nested_lookup import nested_update
 from pathlib import Path
+import logging
+import re
 
 class RecipeUpdate:
     def __init__(self) -> None:
         self.placeholder_keys = []
+        self.placeholder_keys_all = []
+        self.placeholder_keys_any = []
         self.placeholders =[]
         super().__init__()
 
-    def find_placeholders(self, data):
-        self._data = data
-        if isinstance(self._data, dict):
-            for key, value in self._data.items():
-                if isinstance(value, str) and '{{' in value and '}}' in value:
-                    self.placeholder_keys.append(key)
+    def find_placeholders(self, recipe_dict):
+        if isinstance(recipe_dict, dict):
+            for key, value in recipe_dict.items():
+                if isinstance(value, str) and re.search(r"^{{float+.+?all}}$", value):
```

Comment:
Since we might use other data types beside float such as int, str or bool. I would first search for "{{*}}" first and then we take the string inside and split using "." delimiter before using a structure similar to switch-case to parse the data type and whether we will use all values or only the first one

Reply:
Just one doubt regarding this, Do we need to create cases to look for datatype? because irrespective of data type, the _update_recipe() function just matches the key and copies whatever the values to the recipe! if we create a cases for each datatype (int, float, bool and str) all the cases will still be doing the same steps.

Reply:
yes, I see your point. However, the optimiser might return a value of 1.0, which will be added to the recipe. Afterwards, when processing the recipe, this value will cause a problem with mongoengine, as it will throw an error since it expects it to be an int but it was parsed as a float from the recipe